azurerm/resource_arm_iothub_endpoint_storage_container.go

```diff
@@ -0,0 +1,303 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmIotHubEndpointStorageContainer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmIotHubEndpointStorageContainerCreateUpdate,
+		Read:   resourceArmIotHubEndpointStorageContainerRead,
+		Update: resourceArmIotHubEndpointStorageContainerCreateUpdate,
+		Delete: resourceArmIotHubEndpointStorageContainerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.IoTHubEndpointName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"iothub_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.IoTHubName,
+			},
+
+			"container_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.StorageContainerName,
+			},
+
+			"file_name_format": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  false,
+			},
+
+			"batch_frequency_in_seconds": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      300,
+				ValidateFunc: validation.IntBetween(60, 720),
+			},
+
+			"max_chunk_size_in_bytes": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      314572800,
+				ValidateFunc: validation.IntBetween(10485760, 524288000),
+			},
+
+			"connection_string": {
+				Type:     schema.TypeString,
+				Required: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					accountKeyRegex := regexp.MustCompile("AccountKey=[^;]+")
+
+					maskedNew := accountKeyRegex.ReplaceAllString(new, "AccountKey=****")
+					return (new == d.Get(k).(string)) && (maskedNew == old)
+				},
+				Sensitive: true,
+			},
+
+			"encoding": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
Is this required? if so does that mean there is an API or SDK bug where its returning a different constant then in the SDK and should we open an issue?

Reply:
Good call. I removed it and ran through the options and everything looked fine.